### ISSUE TITLE
delete: Unable to get field named "last(TRUE) OVER ...

### ISSUE THREAD

**Issue by BohuTANG** [datafuselabs/databend#13366]
**Summary**

```
2023-10-20T22:27:33.257923Z  INFO databend_sqlsmith::runner: delete_sql: DELETE FROM t1 WHERE last(TRUE) OVER (PARTITION BY NULL, NULL ORDER BY NULL, NULL ASC NULLS FIRST ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) 
panicked at 'called `Result::unwrap()` on an `Err` value: BadArguments. Code: 1006, Text = Unable to get field named "last(TRUE) OVER (PARTITION BY NULL, NULL ORDER BY NULL, NULL ASC NULLS FIRST ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)". Valid fields: ["ccljagv", "c0", "c1", "c2", "c3", "c4", "c5", "c6", "c7", "c8", "c9", "c10", "c11", "c12", "c13", "c14", "c15", "c16", "c17", "c18", "c19", "c20", "c21", "c22", "c23", "c24", "c25", "c26", "c27", "c28", "c29", "c30", "c31", "c32", "c33", "c34", "c35", "c36", "c37"].

<Backtrace disabled by default. Please use RUST_BACKTRACE=1 to enable> ', src/query/storages/fuse/src/operations/delete.rs:145:62
   0: common_tracing::panic_hook::log_panic
             at /workspace/src/common/tracing/src/panic_hook.rs:35:21
   1: common_tracing::panic_hook::set_panic_hook::{{closure}}
             at /workspace/src/common/tracing/src/panic_hook.rs:30:9
   2: call<(&core::panic::panic_info::PanicInfo), (dyn core::ops::function::Fn<(&core::panic::panic_info::PanicInfo), Output=()> + core::marker::Send + core::marker::Sync), alloc::alloc::Global>
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/alloc/src/boxed.rs:1999:9
   3: rust_panic_with_hook
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/std/src/panicking.rs:709:13
   4: {closure#0}
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/std/src/panicking.rs:597:13
   5: __rust_end_short_backtrace<std::panicking::begin_panic_handler::{closure_env#0}, !>
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/std/src/sys_common/backtrace.rs:151:18
   6: begin_panic_handler
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/std/src/panicking.rs:593:5
   7: panic_fmt
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/panicking.rs:67:14
   8: unwrap_failed
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/result.rs:1651:5
   9: core::result::Result<T,E>::unwrap
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/result.rs:1076:23
  10: common_storages_fuse::operations::delete::<impl common_storages_fuse::fuse_table::FuseTable>::try_eval_const::{{closure}}
             at /workspace/src/query/storages/fuse/src/operations/delete.rs:145:40
  11: common_expression::expression::Expr<Index>::project_column_ref
             at /workspace/src/query/expression/src/expression.rs:310:21
  12: common_expression::expression::Expr<Index>::project_column_ref::{{closure}}
             at /workspace/src/query/expression/src/expression.rs:337:46
  13: core::iter::adapters::map::map_fold::{{closure}}
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/iter/adapters/map.rs:84:28
  14: core::iter::traits::iterator::Iterator::fold
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/iter/traits/iterator.rs:2481:21
  15: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/iter/adapters/map.rs:124:9
  16: core::iter::traits::iterator::Iterator::for_each
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/iter/traits/iterator.rs:856:9
  17: alloc::vec::Vec<T,A>::extend_trusted
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/alloc/src/vec/mod.rs:2827:17
  18: <alloc::vec::Vec<T,A> as alloc::vec::spec_extend::SpecExtend<T,I>>::spec_extend
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/alloc/src/vec/spec_extend.rs:26:9
  19: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/alloc/src/vec/spec_from_iter_nested.rs:62:9
  20: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/alloc/src/vec/spec_from_iter.rs:33:9
  21: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/alloc/src/vec/mod.rs:2695:9
  22: core::iter::traits::iterator::Iterator::collect
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/core/src/iter/traits/iterator.rs:1895:9
  23: common_expression::expression::Expr<Index>::project_column_ref
             at /workspace/src/query/expression/src/expression.rs:337:23
  24: common_storages_fuse::operations::delete::<impl common_storages_fuse::fuse_table::FuseTable>::try_eval_const
             at /workspace/src/query/storages/fuse/src/operations/delete.rs:[143](https://github.com/datafuselabs/databend/actions/runs/6593110776/job/17915513311#step:4:144):22
  25: common_storages_fuse::operations::delete::<impl common_storages_fuse::fuse_table::FuseTable>::fast_delete::{{closure}}::{{closure}}
             at /workspace/src/query/storages/fuse/src/operations/delete.rs:117:16
  26: <async_backtrace::framed::Framed<F> as core::future::future::Future>::poll::{{closure}}
             at /opt/rust/cargo/registry/src/index.crates.io-6f17d22bba[150](https://github.com/datafuselabs/databend/actions/runs/6593110776/job/17915513311#step:4:151)01f/async-backtrace-0.2.5/src/framed.rs:49:27
  27: async_backtrace::frame::Frame::in_scope::{{closure}}
             at /opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/async-backtrace-0.2.5/src/frame.rs:184:17
  28: std::thread::local::LocalKey<T>::try_with
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/std/src/thread/local.rs:270:16
  29: std::thread::local::LocalKey<T>::with
             at /rustc/8b35c0bb0f833c0077dc57006eb317edde2a2d1e/library/std/src/thread/local.rs:246:9
  30: async_backtrace::frame::active_frame::with
             at /opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/async-backtrace-0.2.5/src/frame.rs:81:9
  31: async_backtrace::frame::Frame::in_scope
             at /opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/async-backtrace-0.2.5/src/frame.rs:[180](https://github.com/datafuselabs/databend/actions/runs/6593110776/job/17915513311#step:4:181):13
  32: <async_backtrace::framed::Framed<F> as core::future::future::Future>::poll
             at /opt/rust/cargo/registry/src/index.crates.io-6f17d22bba15001f/async-backtrace-0.2.5/src/framed.rs:49:9
  33: common_storages_fuse::operations::delete::<impl common_storages_fuse::fuse_table::FuseTable>::fast_delete::{{closure}}
             at /workspace/src/query/storages/fuse/src/operations/delete.rs:70:5
  34: <databend_query::interpreters::interpreter_delete::DeleteInterpreter as databend_query::interpreters::interpreter::Interpreter>::execute2::{{closure}}
             at /workspace/src/query/service/src/interpreters/interpreter_delete.rs:211:14
```

https://github.com/datafuselabs/databend/actions/runs/6593110776/job/17915513311#step:4:135

**Comment by BohuTANG** [comment]
cc @zhyass

**Comment by b41sh** [comment]
This issue was fixed after PR #13803 merged, the window function is not allowed in the delete where clause.
